vsintegration/src/FSharp.Editor/Common/RoslynHelpers.fs

```diff
@@ -103,8 +103,14 @@ module internal RoslynHelpers =
                   async { do! Async.SwitchToThreadPool()
                           return! computation }, 
                   continuation=(fun result -> disposeReg(); tcs.TrySetResult(result) |> ignore), 
-                  exceptionContinuation=(function :? OperationCanceledException -> disposeReg(); tcs.TrySetCanceled(cancellationToken)  |> ignore
-                                                | exn -> disposeReg(); tcs.TrySetException(exn) |> ignore),
+                  exceptionContinuation=(function :? OperationCanceledException -> 
+                                                        disposeReg()
+                                                        if tcs.Task.Status = TaskStatus.Running then
+                                                            tcs.TrySetCanceled(cancellationToken)  |> ignore
+                                                | exn ->
+                                                    disposeReg()
+                                                    if tcs.Task.Status = TaskStatus.Running then
```

Comment:
I can see cancelling the task when it is not still running is a problem.  But shouldn't we call set exception whether the task is running or not?